src/api.js

```diff
@@ -844,6 +912,48 @@ Module["onRuntimeInitialized"] = function onRuntimeInitialized() {
         return stmt;
     };
 
+    /** Iterate over multiple SQL statements in a SQL string
+     @param {string} sql a string of SQL
+     @return {StatementIterator} the resulting statement iterator
+     */
+    Database.prototype["iterateStatements"] = function iterateStatements(sql) {
+        return new StatementIterator(sql, this);
+    };
+
+    /* Internal method to implement statement iteration */
+
+    Database.prototype["advanceIterator"] = function advanceIterator(iter) {
```

Comment:
We don't want to make this public (no `[""]`). And this should probably not be a method on Database.

Reply:
OK on the public - I didn't know how that worked, should be an easy change.  As for it being a method on Database, it felt like it really should belong there, given that all similar code (Database.prepare, Database.exec) is in Database.  I actually considered not making a StatementIterator class at all, in favor of a simple object, since I can delegate all the important code to Database.  Having it in Database may also provide opportunities for future refactoring.

In addition, the method does access what I think of as "private" members of the Database object.

Reply:
Ok for keeping that on the database then.

Reply:
The code now doesn't access any private code in Database, but it accesses private fields in StatementIterator. Let's move it to StatementIterator.next

Reply:
I see what you mean about using private fields in StatementIterator, but just to double check, what about the line (in advanceIterator()):
```
this.statements[pStmt] = iter.activeStatement;
```
?

In StatementIterator.next() this would become
```
this.db.statements[pStmt] = this.activeStatement;
```

Database::statements seems like "private" data to me.

I don't know how it would be done in Javascript, but if I were in C++, I would want to make StatementIterator a private class within Database.  Then it would be fine that it is accessing private member variables of Database.

Reply:
There is no notion of visibility in javascript, but I think it's okay to consider Database::statements visible at the level of the library.